src/approaches/interactive_learning_approach.py

```diff
@@ -21,95 +23,116 @@ class InteractiveLearningApproach(NSRTLearningApproach):
-            preds = self._get_current_predicates() - static_preds
-            # Find policy for exploration
-            task_list = glib_sample(state, preds, dataset)
-            assert task_list
-            task = task_list[0]
-            for task in task_list:
-                try:
-                    print("Solving for policy...")
-                    policy = self.solve(task, timeout=CFG.timeout)
-                    break
-                except (ApproachTimeout, ApproachFailure) \
-                        as e:  # pragma: no cover
-                    print(f"Approach failed to solve with error: {e}")
+        # Save dataset, to be used for online interaction.
+        self._dataset = dataset
+
+    def get_interaction_requests(self) -> List[InteractionRequest]:
+        # Sample a train task.
+        train_task_idx = self._rng.choice(len(self._train_tasks))
+        init = self._train_tasks[train_task_idx].init
+        # Detect and filter out static predicates.
+        static_preds = utils.get_static_preds(self._nsrts,
+                                              self._predicates_to_learn)
+        preds = self._predicates_to_learn - static_preds
+        # Find acting policy for the request.
+        task_list = glib_sample(init, preds, self._dataset)
+        assert task_list
+        task, act_policy = self._find_first_solvable(task_list)
+        assert task.init is init
+
+        def _query_policy(s: State) -> Optional[GroundAtomsHoldQuery]:
+            # Decide whether to ask about each possible atom.
+            ground_atoms = utils.all_possible_ground_atoms(
+                s, self._predicates_to_learn)
+            atoms_to_query = set()
+            for atom in ground_atoms:
+                score = score_atom(self._dataset, atom)
+                # Ask about this atom if it is the best seen so far.
+                if score > self._best_score:
+                    atoms_to_query.add(atom)
+                    self._best_score = score
+            if not atoms_to_query:
+                return None
+            return GroundAtomsHoldQuery(atoms_to_query)
+
+        def _termination_function(s: State) -> bool:
+            # Stop the episode if we reach the goal that we babbled.
+            return all(goal_atom.holds(s) for goal_atom in task.goal)
+
+        return [
+            InteractionRequest(train_task_idx, act_policy, _query_policy,
+                               _termination_function)
+        ]
+
+    def learn_from_interaction_results(
+            self, results: Sequence[InteractionResult]) -> None:
+        assert len(results) == 1
+        result = results[0]
+        for state, response in zip(result.states, result.responses):
+            if response is None:
+                continue  # we didn't ask a query on this timestep
+            assert isinstance(response, GroundAtomsHoldResponse)
+            assert isinstance(response.query, GroundAtomsHoldQuery)
```

Comment:
remove this line, we no longer need it (it was initially there to appease mypy when we were using `response.query`)